### PR TITLE
Save default Z Index | Multi Z dev doc

### PR DESCRIPTION
**Pull request by rastogi-bhavya**
### Save Default Z
A UI has been added which allows the user to update the default z index.

#### Working
 - Based on the value of `canUpdateDefaultZIndex` the button to update the default z index would be shown to the user.
 - When the user switches to an index which is not the default Z, the `Save` button will become active. If the user decides to update the default value, he will have to click this button.
 - When this happens a spinner will appear in place of the button indicating that the request is being processed.
 - After the request is processed successfully, the button will go back to a disabled state, until switch to a different z index.

### Dev Doc
A dev doc has been added explaining how the features of multi z are working. This doc is however not complete yet.

### CODE REVIEW

**Comment by RFSH** [comment]
@rastogi-bhavya Since the button is small, the icon is taking up most of it and is not easily distinguishable. So please try making the button bigger. We might also investigate changing the icon to be something else.

File: js/toolbar/z-plane-list.js [hunk 1]
```diff
@@ -426,11 +485,28 @@ function ZPlaneList(parent) {
      */
     this._renderZPlaneInfo = function () {
         var zPlaneInfo = document.getElementById('z-plane-info-container');
+
+
+        if (_self.canUpdateDefaultZIndex == true) {
```
Comment (RFSH) [inline js/toolbar/z-plane-list.js:490]: Improve this part. It's ugly. You don't need `last-button` class. You can select the last child in css [like this](https://developer.mozilla.org/en-US/docs/Web/CSS/:last-child). Then you don't need to repeat the `jumpButton`.

File: js/toolbar/z-plane-list.js [hunk 2]
```diff
@@ -97,12 +97,16 @@ function ZPlaneList(parent) {
...
 
+    this.defaultZIndex = -1;
+
     /**
      * called on load to calculate the page size and then ask chaise to fetch the results
      * @param {object} data
      */
     this.init = function (data) {
 
+        // TODO use canUpdateDefaultZIndex
```
Comment (RFSH) [inline js/toolbar/z-plane-list.js:108]: remove this

**Comment by RFSH** [comment]
I should mention that I pushed two changes to this PR:

- Downgraded bootstrap so glyphicons can be used. We're not using bootstrap directly so there really wasn't any need to upgrade it in the first place (I'm assuming at the time that upgrade happened bootstrap was used but it's not the case anymore)
- Pulled master into this branch so proper tooltips can be used. And also modified `z-plane-list.js` to use the new tooltip mechanism.

**Comment by rastogi-bhavya** [comment]
I have updated the button used for search and update of z index, and also the spinner.

File: js/app.js [hunk 3]
```diff
@@ -173,6 +176,10 @@ var OSDViewer = (function (_config) {
                 case 'updateZPlaneList':
                     toolbar && toolbar.updateZPlaneList(data);
                     break;
+                case 'updateDefaultZIndexDone':
+                    // TODO
```
Comment (RFSH) [inline js/app.js:180]: remove this

File: js/toolbar/z-plane-list.js [hunk 4]
```diff
@@ -97,6 +97,8 @@ function ZPlaneList(parent) {
      */
     this.validFetchRequests = ['fetchZPlaneListByZIndex', 'fetchZPlaneList'];
 
+    this.defaultZIndex = -1;
```
Comment (RFSH) [inline js/toolbar/z-plane-list.js:100]: comment what this is (like the previous ones)

```
/**
* DESCRIPTION GOES HERE
* @type {integer}
*/
```

File: js/toolbar/z-plane-list.js [hunk 5]
```diff
@@ -127,15 +129,24 @@ function ZPlaneList(parent) {
...
         _self._render();
         _self._fetchListByZIndex('default');
     }
 
     /**
-     * TODO list all the properties of the `data` and explain each one.
      * @param {object} data
      */
     this.updateList = function (data) {
+        // Properties of data
```
Comment (RFSH) [inline js/toolbar/z-plane-list.js:142]: Move this comment to be above the function in the jsdoc format

File: js/toolbar/z-plane-list.js (hunk 4 above)
Comment (rastogi-bhavya) [reply js/toolbar/z-plane-list.js:104]: Done

File: js/app.js (hunk 3 above)
Comment (rastogi-bhavya) [reply js/app.js]: Done

**Review by RFSH — APPROVED** [review]

**Pull request merged**